### PR TITLE
Fix openssl detection on win64

### LINKED ISSUE #2031: Can't update card database over SSL
<b>OS:</b>
Windows 7

<b>Cockatrice version:</b>
c19f225 (2016-06-26)

-___
<br>

When I open cockatrice, a window says, "Cockatrice is unable to load the card database. Do you want to update your card database now? If unsure or first time user, choose "Yes"

I press "Yes". It brings me to language choice screen, I select english and press "next". Now is the source selection screen. The "Download URL:" option has "https://mtgjson/json/AllSets.json.zip" as the source. I click next. Then a new window immediately pops up saying, "Network error: Error creating SSL context ()."

How do I move past this?

### PR DESCRIPTION
**Pull request by ctrlaltca**
Fix #2031

### CODE REVIEW

**Comment by tooomm** [comment]
Working for me now! :+1:

**Comment by ctrlaltca** [comment]
Ready for merge when the CI turns green, then

File: cmake/FindWin32SslRuntime.cmake [hunk 1]
```diff
@@ -5,22 +5,37 @@ if (WIN32)
...
+      "[HKEY_LOCAL_MACHINE\\SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\Uninstall\\OpenSSL (64-bit)_is1;Inno Setup: App Path]"
+      ENV OPENSSL_ROOT_DIR
+      )
+    file(TO_CMAKE_PATH "$ENV{PROGRAMFILES}" _programfiles)
+    set(_OPENSSL_ROOT_PATHS
+      "${_programfiles}/OpenSSL-Win64"
+      "C:/OpenSSL-Win64/"
+      )
+    unset(_programfiles)
+  else( CMAKE_SIZEOF_VOID_P EQUAL 8 )
```
Comment (Daenyth) [inline cmake/FindWin32SslRuntime.cmake]: This is the same if clause as above?

```
if( CMAKE_SIZEOF_VOID_P EQUAL 8 )
else( CMAKE_SIZEOF_VOID_P EQUAL 8 )
```

File: cmake/FindWin32SslRuntime.cmake [hunk 2]
```diff
@@ -33,6 +48,7 @@ set(_OPENSSL_ROOT_HINTS_AND_PATHS
     PATHS ${_OPENSSL_ROOT_PATHS}
     )
 
+# Even if the dll is 64bit, it's still suffixed as *32.dll
```
Comment (Daenyth) [inline cmake/FindWin32SslRuntime.cmake]: Goddamnit windows

File: cmake/FindWin32SslRuntime.cmake (hunk 1 above)
Comment (Daenyth) [inline cmake/FindWin32SslRuntime.cmake]: oh I see it's else, not elseif..

Comment (ctrlaltca) [inline cmake/FindWin32SslRuntime.cmake]: They guys developing cmake originally made some weird syntax choices, like this one. We already define `CMAKE_ALLOW_LOOSE_LOOP_CONSTRUCTS`, so we could remove the condition from else() and endif()

**Pull request merged**